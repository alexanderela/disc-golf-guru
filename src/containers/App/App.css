.App {
  text-align: center;
  background-color: #E9EBE5;
  padding: 0 0 7rem 0;
}


