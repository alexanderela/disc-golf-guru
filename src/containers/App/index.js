import React, { Component } from 'react';
import './App.css';
import Header from '../../components/Header';
import CardContainer from '../../components/CardContainer';
import Nav from '../Nav';
import MainPage from '../MainPage';
import { Route, Switch } from 'react-router-dom';
import Home from '../../components/Home';
import Error from '../../components/Error';
import { withRouter } from 'react-router';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';


class App extends Component {
  constructor() {
    super();
    this.state = {
      favoriteClicked: ''
    }
  }

  updateFavorites = (favoriteCourse) => {
    this.setState({ favoriteClicked: favoriteCourse })
  }

  getLocalStorage = categoryName => {
    if (localStorage) {
      return JSON.parse(localStorage.getItem(categoryName));
    } else {
      return
    }
  };

  checkLocalStorage = () => {
    if (localStorage.favorites) {
      const retrievedFavorites = this.getLocalStorage('favorites');
      return retrievedFavorites;
    } else {
      return [];
    }
  };

  render() {
    const filteredFavorites = this.checkLocalStorage()
    const { favoriteClicked } = this.state

    return (
      <div className="App">
        <Header />
        <Nav />
        <Switch>
          <Route exact path="/disc-golf-guru" render={() => <Home />} />
          <Route
            path="/findcourses"
            render={() => <MainPage 
              pageName={'Find A Disc Golf Course'} 
              updateFavorites={this.updateFavorites}
              />}
          />
          <Route
            path="/favorites"
            render={() =>
              localStorage.favorites ? (
                <CardContainer 
                  favorites={filteredFavorites} 
                  updateFavorites={this.updateFavorites}
                  favoriteSelected={favoriteClicked}                />
              ) : (
                <Error message={'You currently have no favorites selected'} />
              )
            }
          />
          <Route
            path="*"
            exact={true}
            render={() => (
              <Error message="The page you are looking for does not exist." />
            )}
          />
        </Switch>
      </div>
    );
  }
}

export const mapStateToProps = ({ golfCourses }) => ({ golfCourses });

App.propTypes = {
  golfCourses: PropTypes.array.isRequired
}

export default withRouter(
  connect(
    mapStateToProps,
    null
  )(App)
);
