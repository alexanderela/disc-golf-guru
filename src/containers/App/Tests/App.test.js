import React from 'react';
import ReactDOM from 'react-dom';
import App from '../';
import { shallow } from 'enzyme';

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  afterEach(() => {
		localStorage.clear()  	
  })

  it('should render like snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('getLocalStorage', () => {
		it('gets and parses data from local storage', () => {
			const mockData = [{ "class": "wheeled", "model": "Digger Crawler" }, 
			{"class": "tires", "model": "Mustang"}]
			localStorage.setItem('mockData', JSON.stringify(mockData))
			const	getStorage = wrapper.instance().getLocalStorage('mockData')
			expect(getStorage).toEqual(mockData) 
		})
  })

  describe('checkLocalStorage', () => {
		// it('retrieves planets from local storage if there are planets in local storage', async () => {
		// 		localStorage.setItem('planets', JSON.stringify(mockVehicle))
		// 		const spy = spyOn(wrapper.instance(), 'getLocalStorage')	
		// 		wrapper.instance().showPlanets()	
		// 		expect(spy).toHaveBeenCalled()				
		// 	})
  	
	})
});
