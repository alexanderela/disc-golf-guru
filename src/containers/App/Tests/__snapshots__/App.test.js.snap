// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render like snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header />,
        <Nav />,
        <Switch>
          <Route
            exact={true}
            path="/"
            render={[Function]}
          />
          <Route
            path="/findcourses"
            render={[Function]}
          />
          <Route
            path="/favorites"
          />
        </Switch>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />,
            <Route
              path="/findcourses"
              render={[Function]}
            />,
            <Route
              path="/favorites"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/findcourses",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/favorites",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header />,
          <Nav />,
          <Switch>
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />
            <Route
              path="/findcourses"
              render={[Function]}
            />
            <Route
              path="/favorites"
            />
          </Switch>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                exact={true}
                path="/"
                render={[Function]}
              />,
              <Route
                path="/findcourses"
                render={[Function]}
              />,
              <Route
                path="/favorites"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/findcourses",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/favorites",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
